pkg/providers/instancetype/types.go

```diff
@@ -132,6 +132,17 @@ func computeRequirements(ctx context.Context, info *ec2.InstanceTypeInfo, offeri
 	return requirements
 }
 
+func getOS(info *ec2.InstanceTypeInfo, amiFamily amifamily.AMIFamily) string {
```

Comment:
I think we need to process this requirement a bit differently. If we return an empty string here e.g. `""`. That means that the scheduling requirement says that the instance type supports an OS with the value of `""`. Rather, I think what we want to say is that it doesn't support this OS at all.

We may need to modify this function to return the OS scheduling requirement like

```go
func getOSRequirement(info *ec2.InstanceTypeInfo, amiFamily amifamily.AMIFamily) string {
   if _, ok := amiFamily.(*amifamily.Windows); ok {
	if getArchitecture(info) == v1alpha5.ArchitectureAmd64 {
		return scheduling.NewRequirement(v1.LabelOSStable, v1.OperatorIn, string(v1.Windows)
	}
        return scheduling.NewRequirement(v1.LabelOSStable, DoesNotExist)
  }
  return scheduling.NewRequirement(v1.LabelOSStable, v1.OperatorIn, string(v1.Linux))
}

Reply:
sure!